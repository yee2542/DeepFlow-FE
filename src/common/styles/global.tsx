import 'antd/dist/antd.css';
import styled, { createGlobalStyle, css } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  html {
    font-size: 100%;
  }

  :root {
      //COLORS
    --primary-color: #4425A7;
    --secondary-color: #5A36CC;
    --link-color: #4425A7;
    --success-color: #27AE60;
    --light-success-color: #6fcf97;
    --warning-color: #FF8863;
    --error-color: #EB5757;
    --light-error-color: #f07979;
    --tertiary-color: #E1D8FF;
    --green-color: #0BB5C2;
    --light-green-color: #48cdd8;
    --red-color: #EB5757;
    --light-gray-color: lightgray;
    --gray-color: #F2F2F2;
    --dark-gray-color: #BDBDBD;

    //FONTS
    --font-size-base: 16px;
    --font-size-sm: 14px
    --font-size-lg: 18px
    --text-color: #4F4F4F;
    --text-color-secondary: #88888D;
    --h1: 3.052rem;
    --h2:2.441rem;
    --h3:1.953rem;
    --h4:1.563rem;
    --h5: 1.25rem;
    --body: 16px; 
    --small-body: 0.8rem;
    --caption: 0.64rem;
    
    --box-shadow-base: rgba(149, 157, 165, 0.2) 0px 8px 24px;
    --border-radius-base: 8px;
    --border-radius-round: 100px;

    // LAYOUT BREAKPOINT 
    --xs: 480px;
    --sm: 576px;
    --md: 768px;
    --lg: 992px;
    --xl: 1200px;
    --xxl: 1600px;
  }


  body {
    padding: 0 !important;
    margin: 0 !important;
    font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen,
      Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
  }

  hr {
    border: none;
    background-color: var(--light-gray-color);
    height: 1px;
  }

  h1 {
    font-size: var(--h1);
    font-weight: bold;
    color: var(--text-color);
  }

  h2 {
    font-size: var(--h2);
    font-weight: bold;
    color: var(--text-color);
  }

  h3 {
    font-size: var(--h3);
    font-weight: bold;
    color: var(--text-color);
  }

  h4 {
    font-size: var(--h4);
    font-weight: bold;
    color: var(--text-color);
  }

  h5 {
    font-size: var(--h5);
    font-weight: bold;
    color: var(--text-color);
  }

  span,p, body {
    font-size: var(--body);
    color: var(--text-color);
  }
`;

interface Align {
  type?: 'center' | 'left' | 'right';
}

export type PComponentProps = {
  align?: Align['type'];
} & React.DetailedHTMLProps<
  React.HTMLAttributes<HTMLParagraphElement>,
  HTMLParagraphElement
>;

const handleAlignType = (align?: PComponentProps['align']) => {
  switch (align) {
    case 'right':
      return css`
        text-align: right;
      `;
    case 'left':
      return css`
        text-align: left;
      `;
    default:
      // 'center
      return css`
        text-align: center;
      `;
  }
};

export const P = styled.p<PComponentProps>`
  ${({ align }) => handleAlignType(align)};
  color: darkgray;
`;
