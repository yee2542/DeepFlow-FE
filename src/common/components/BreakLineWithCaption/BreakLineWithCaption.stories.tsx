import { Meta, Story } from '@storybook/react';
import { BreakLineWithCaption } from 'Components/BreakLineWithCaption';
import { GlobalStyle } from 'Styles/global';
import React from 'react';

export default {
  title: 'FluentSearch/BreakLineWithCaption',
  component: BreakLineWithCaption,
} as Meta;

type PropType = {
  children?: HTMLCollection | string;
};

const BreakLineWithCaptionStory: Story<PropType> = (args) => (
  <div style={{ marginTop: '30px' }}>
    <GlobalStyle />
    <BreakLineWithCaption {...args} />
  </div>
);

export const Default = BreakLineWithCaptionStory.bind({});
Default.args = {
  children: 'or',
};
