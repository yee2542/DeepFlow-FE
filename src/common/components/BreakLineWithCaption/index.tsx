import React from 'react';
import { Row, Col } from 'antd';
import { P } from 'Styles/global';

export const BreakLineWithCaption: React.FC = (props) => {
  return (
    <Row justify="center">
      <Col span={10}>
        <hr />
      </Col>
      <Col span={4}>
        <P>{props.children}</P>
      </Col>
      <Col span={10}>
        <hr />
      </Col>
    </Row>
  );
};
