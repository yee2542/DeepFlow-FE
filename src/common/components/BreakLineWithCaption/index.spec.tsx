import { shallow } from 'enzyme';
import React from 'react';
import { BreakLineWithCaption } from 'Components/BreakLineWithCaption/index';
import { P } from 'Styles/global';

describe('Component/BreakLineWithCaption test', () => {
  it('Render correctly', () => {
    const wrap = shallow(<BreakLineWithCaption />);
    expect(wrap).toMatchSnapshot();
  });

  it('BreakLineWithCaption isString render test', () => {
    const wrapWithString = shallow(<BreakLineWithCaption> or </BreakLineWithCaption>);
    const wrapWithDiv = shallow(
      <BreakLineWithCaption>
        <div>or</div>
      </BreakLineWithCaption>,
    );

    expect(wrapWithString.exists(P)).toBe(true);
    expect(wrapWithString.exists('div')).toBe(false);
    expect(wrapWithDiv.exists('div')).toBe(true);
    expect(wrapWithDiv.exists(P)).toBe(false);
  });
});
