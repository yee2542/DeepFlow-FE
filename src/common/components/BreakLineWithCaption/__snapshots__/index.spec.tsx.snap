// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/BreakLineWithCaption test Render correctly 1`] = `ShallowWrapper {}`;
